core/src/main/java/io/questdb/griffin/engine/functions/groupby/FirstNotNullCharGroupByFunction.java

```diff
@@ -37,14 +38,29 @@ public FirstNotNullCharGroupByFunction(@NotNull Function arg) {
     }
 
     @Override
-    public void computeNext(MapValue mapValue, Record record) {
-        if (mapValue.getChar(valueIndex) == CharConstant.ZERO.getChar(null)) {
-            computeFirst(mapValue, record);
+    public void computeNext(MapValue mapValue, Record record, long rowId) {
```

Comment:
there is no compute first implementation here, how do we arrive to the conclusion that 

```
mapValue.getChar(valueIndex + 1) == CharConstant.ZERO.getChar(null)
```

Reply:
i see, there is inheritance based behaviour. Can we make it more explicit? It looks as if you get first value if column does not have null values. Either way, not very clear what's going on

Reply:
We use inheritance across all first/last functions. Alternatively, we could duplicate the code or extract common code into abstract classes. WDYT?